Automattic/wp-calypso
# Limited Global Styles: Generalize badges for custom styles

Fixes https://github.com/Automattic/dotcom-forge/issues/2856
Fixes https://github.com/Automattic/wp-calypso/issues/78638

## Proposed Changes

- Changes the theme type badges so _locked_ style variations are labeled with "★ Upgrade" instead of "★ Premium" which might incorrectly infer they need a Premium plan.
- For the same reason, the "Premium style" title has been removed from the tooltip, since just the tooltip message should be enough.
- Additionally, this also fixes the badges in the Theme showcase so _locked_ style variations are correctly labeled.

Section | Before | After
--- | --- | ---
Design picker | <img width="590" alt="Screenshot 2023-06-23 at 13 11 33" src="https://github.com/Automattic/wp-calypso/assets/1233880/0911dca2-9649-4cf1-8147-fba63a93e564"> | <img width="598" alt="Screenshot 2023-06-28 at 10 55 54" src="https://github.com/Automattic/wp-calypso/assets/1233880/0329e25d-54f2-4168-be63-658d726f1893">
Theme showcase | <img width="418" alt="Screenshot 2023-06-28 at 10 56 49" src="https://github.com/Automattic/wp-calypso/assets/1233880/81a5222b-dfda-4951-8b58-02a8edab419d"> | <img width="505" alt="Screenshot 2023-06-28 at 10 57 16" src="https://github.com/Automattic/wp-calypso/assets/1233880/58c6e2af-6064-4a64-a71e-6d5223a797c7">


## Testing Instructions

- Use the Calypso live link below.
- Go to `/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>`.
- Find a free theme with style variations.
- Click on a disc that represents a custom style variation.
- For sites with Limited Global Styles, make sure the badge changes to "★ Upgrade" 
- For sites with full Global Styles, make sure it remains as "Free".
- For sites with Limited Global Styles, make sure the tooltip no longer displays "Premium style" in the title.
- For sites with full Global Styles, make sure there is no tooltip.
- Find now a premium theme with style variations.
- Click on a disc that represents a custom style variation.
- Make sure the badge is still "★ Premium".
- Make sure the tooltip title still says "Premium theme".
- Repeat the steps in the theme showcase (both while logged-in and logged-out, but note that you cannot full GS if logged-out).
